presto-native-execution/presto_cpp/main/types/PrestoToVeloxSplit.cpp

```diff
@@ -91,6 +101,34 @@ velox::exec::Split toVeloxSplit(
             serdeParameters),
         splitGroupId);
   }
+
+  if (auto icebergSplit =
+          std::dynamic_pointer_cast<const protocol::IcebergSplit>(
+              connectorSplit)) {
+    std::unordered_map<std::string, std::optional<std::string>> partitionKeys;
+    for (const auto& entry : icebergSplit->partitionKeys) {
+      partitionKeys.emplace(
+          entry.second.name,
+          entry.second.value == nullptr
+              ? std::nullopt
+              : std::optional<std::string>{*entry.second.value});
+    }
+    std::unordered_map<std::string, std::string> customSplitInfo;
+
+    return velox::exec::Split(
+        std::make_shared<connector::hive::HiveConnectorSplit>(
+            scheduledSplit.split.connectorId,
+            icebergSplit->path,
+            toVeloxFileFormat(icebergSplit->fileFormat),
+            icebergSplit->start,
+            icebergSplit->length,
+            partitionKeys,
+            std::nullopt,
```

Comment:
The last three parameters are optional, right?

Reply:
Oh yes, missed that, removing these.